__tests__/integration/atomic-swap.test.js

```diff
@@ -0,0 +1,305 @@
+import { TestUtils } from './utils/test-utils-integration';
+import { WalletHelper } from './utils/wallet-helper';
+import { singleMultisigWalletData } from '../../scripts/helpers/wallet-precalculation.helper';
+import { loggers } from './utils/logger.util';
+
+describe('send tx (HTR)', () => {
+  let wallet1;
+  let wallet2;
+  let walletMultisig;
+
+  let tokenTx1;
+  let tokenTx2;
+
+  const {
+    words: multisigWords,
+    walletConfig: multisigWalletConfig,
+  } = singleMultisigWalletData;
+
+  beforeAll(async () => {
+    global.config.seeds = { multisig: multisigWords[0] };
+    global.config.multisig = { multisig: multisigWalletConfig };
+    try {
+      wallet1 = WalletHelper.getPrecalculatedWallet('atomic-swap-1');
+      wallet2 = WalletHelper.getPrecalculatedWallet('atomic-swap-2');
+      walletMultisig = new WalletHelper('multisig', { seedKey: 'multisig', multisig: true });
+
+      await WalletHelper.startMultipleWalletsForTest([
+        wallet1,
+        wallet2,
+        walletMultisig,
+      ]);
+
+      // Funds for single input/output tests
+      await wallet1.injectFunds(1000, 0);
+      await wallet2.injectFunds(1000, 0);
+      await walletMultisig.injectFunds(1000, 0);
+
+      tokenTx1 = await wallet1.createToken({ amount: 100, name: 'Token wallet1', symbol: 'TKW1' });
+      tokenTx2 = await wallet2.createToken({ amount: 100, name: 'Token wallet2', symbol: 'TKW2' });
+
+      // Awaiting for updated balances to be received by the websocket
+      await TestUtils.pauseForWsUpdate();
+    } catch (err) {
+      TestUtils.logError(err.stack);
+    }
+  });
+
+  afterAll(async () => {
+    global.config.seeds = {};
+    global.config.multisig = {};
+    await wallet1.stop();
+    await wallet2.stop();
+    await walletMultisig.stop();
+  });
+
+  it('should exchange HTR between 2 wallets', async () => {
+    // wallet1 proposes the transaction
+    let response = await TestUtils.request
+      .post('/wallet/atomic-swap/tx-proposal')
+      .send({ send_tokens: [{ value: 10 }] })
```

Comment:
Thinking about the logical flow in my opinion, the one who proposes the transaction will say that he wants to send 10 HTR and receive 10 HTR, so we can fill the utxos and the outputs, right?